src/client/sandbox/node/index.js

```diff
@@ -85,14 +86,28 @@ export default class NodeSandbox extends SandboxBase {
         // NOTE: In Google Chrome, iframes whose src contains html code raise the 'load' event twice.
         // So, we need to define code instrumentation functions as 'configurable' so that they can be redefined.
         Object.defineProperty(window, INTERNAL_PROPS.processDomMethodName, {
-            value:        (el, doc) => this.processNodes(el, doc),
+            value: (el, doc) => {
+                // NOTE: testcafe creates shadow-ui root before DOMContentLoaded event (once document.body is available).
+                // Sometimes for very heavy DOM or very slow loading at that moment body doesn't contain all elements
```

Comment:
// Sometimes for a very heavy DOM or a very slow loading at that moment, the body doesn't contain all elements